# Домашнее задание. Урок 6.

**1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.**

In [168]:
import numpy as np
import pandas as pd
from scipy import stats

In [169]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [170]:
mean_zp = 1 / len(zp) * sum([x for x in zp])
mean_ks = 1 / len(ks) * sum([y for y in ks])
mean_zp, mean_ks

(101.4, 709.9000000000001)

In [171]:
cov = sum(map(lambda x,y: (x - mean_zp) * (y - mean_ks), zp, ks)) / (len(zp) - 1)

In [172]:
cov

10175.377777777778

In [173]:
a = [x for x in range(len(zp))]

In [174]:
df = pd.DataFrame(list(zip(ks, zp)))

In [175]:
df.cov(ddof=1)

0             1
0  33854.322222  10175.377778
1  10175.377778   3882.933333

In [176]:
np.cov(zp, ks, ddof=1)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

__значения ковариации совпадают во всех случаях__

In [177]:
std_zp = (sum([(x - mean_zp) ** 2 for x in zp]) / (len(zp) - 1)) ** 0.5
std_ks = (sum([(y - mean_ks) ** 2 for y in ks]) / (len(ks) - 1)) ** 0.5
std_zp, std_ks

(62.31318747531162, 183.99544076477065)

In [178]:
r = cov / (std_zp * std_ks)
r

0.8874900920739163

In [179]:
df.corr()[0][1], np.corrcoef(zp, ks)[0][1]

(0.8874900920739165, 0.8874900920739162)

*Ответ: значение ковариации **cov = 10175.4**, а значение корреляции Пирсона признаков **r = 0.8875**.*

**2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.**

In [180]:
samples = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
a = 1 - 0.95

In [181]:
t1 = stats.t.ppf(a/2, df=len(samples) - 1)
t2 = stats.t.ppf(1 - a/2, df=len(samples) - 1)
t1, t2

(-2.2621571627409915, 2.2621571627409915)

In [182]:
se = samples.std(ddof=1) / np.sqrt(len(samples))

In [183]:
mu = samples.mean()
mu

118.1

In [184]:
p1 = mu + t1 * se
p2 = mu + t2 * se
p1, p2

(110.55608365158724, 125.64391634841274)

Ответ:  __[110.556, 125.6439]__

**3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.**

In [185]:
var = 25
std = np.sqrt(var)
n = 27
mu = 174.2
a = 0.05
std

5.0

In [186]:
n1 = stats.norm.ppf(a/2)
n2 = stats.norm.ppf(1 - a/2)
n1, n2

(-1.9599639845400545, 1.959963984540054)

In [187]:
se = std / np.sqrt(n)

In [188]:
p1 = mu + n1 * se
p2 = mu + n2 * se
p1, p2

(172.3140237765397, 176.08597622346028)

Ответ: __[172.3, 176.086]__